workflow/src/main/java/com/synopsys/integration/alert/workflow/message/NotificationReceiver.java

```diff
@@ -51,32 +54,46 @@
 
     @Autowired
     public NotificationReceiver(Gson gson, NotificationAccessor notificationAccessor, NotificationProcessor notificationProcessor, ChannelEventManager eventManager) {
-        super(gson, NotificationEvent.class);
+        super(gson, NotificationReceivedEvent.class);
         this.notificationAccessor = notificationAccessor;
         this.notificationProcessor = notificationProcessor;
         this.eventManager = eventManager;
     }
 
     @Override
-    public void handleEvent(NotificationEvent event) {
-        if (NotificationEvent.NOTIFICATION_EVENT_TYPE.equals(event.getDestination())) {
-            if (null == event.getNotificationIds() || event.getNotificationIds().isEmpty()) {
-                logger.warn("Can not process a notification event without notification Id's.");
-                return;
-            }
-            logger.debug("Event {}", event);
-            logger.info("Processing event for {} notifications.", event.getNotificationIds().size());
-            List<AlertNotificationModel> notifications = notificationAccessor.findByIds(event.getNotificationIds());
+    public void handleEvent(NotificationReceivedEvent event) {
+        if (!NotificationReceivedEvent.NOTIFICATION_RECEIVED_EVENT_TYPE.equals(event.getDestination())) {
+            logger.warn("Received an event of type '{}', but this listener is for type '{}'.", event.getDestination(), NotificationReceivedEvent.NOTIFICATION_RECEIVED_EVENT_TYPE);
+            return;
+        }
```

Comment:
Is this a valid concern? Shouldn't this receiver be setup to only run when this event type is being processed?

Reply:
This was refactored as part of Gavin's earlier suggestion. Looking into it I'm not entirely sure if it would be possible in this setup for the message to go to the wrong place since its the only consumer of NotificationReceivedEvent. In the event there were multiple consumers I believe then it would be a concern but I'm not too sure, I would need to investigate how JMS would handle that. If others think its no longer relevant then I can remove this.